### ISSUE TITLE
UEFI Boot Option fails to start after fresh 19.03 install

### ISSUE THREAD

**Issue by codygman** [NixOS/nixpkgs#70518]
**Describe the bug**
I installed nixos with a USB live copy of NixOS 19.03 following the manual's instructions for UEFI boot.

I then rebooted and because my usb drive was next in my boot order, I came back to the usb live installer.

I ran `efibootmgr` and I saw something like:

```
Linux Boot Manager
USB UEFI...
Windows...
```

So I'm pretty sure a boot loader was installed. I remounted my boot partition to /mnt/boot and observed that /mnt/boot/EFI/systemd existed.

This may be related to #45032.

I can't try the grub solution they posted now, but will soon:

```
# Use the systemd-boot EFI boot loader.
#boot.loader.systemd-boot.enable = true;
boot.loader.grub.enable = true;
boot.loader.grub.device = "nodev";
boot.loader.grub.efiSupport = true;
boot.loader.efi.canTouchEfiVariables = true;
```

**Comment by veprbl** [comment]
@codygman You never explain how things go wrong for you. Is there an error message that you see? At what point?

**Comment by mudrii** [comment]
@codygman 

I am running Nixos on UEFi with below config

```
   boot = {
      loader = {
        systemd-boot.enable = true;
        efi.canTouchEfiVariables = true;
        grub.enableCryptodisk = true;
      };
      kernelPackages = pkgs.linuxPackages_latest;
    };
```

**Comment by codygman** [comment]
My apologies, I was up quite late battling with it and wanted to make sure
my issue was recorded.

It turns out in my BIOS I had set a default within the 'boot from' hard
disk option even though per efibootmgr Linux was first.

I successfully booted with grub though, so I can't yet say sysytemd-boot
works for me.

As soon as possible Ill revert to the installer default, ensure that
'smart' override atop efi boot order is set, and report back.

Probably good to add to FAQ of manual in case others waste 5 hours with
this same silly 'smart' BIOS issue.
